From c6f6cb5f41e553589b7eb5b8913f83e21499b659 Mon Sep 17 00:00:00 2001
From: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
Date: Fri, 20 Jul 2018 00:12:15 -0700
Subject: [PATCH 2062/2367] ASoC: SOF: decrement device usage counter after
 booting firmware

Decrement device usage counter for both the sof device
and the pci/acpi/spi device.
Without this change the runtime_usage count never reaches 0
and the device will not suspend.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
---
 sound/soc/sof/core.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/sound/soc/sof/core.c b/sound/soc/sof/core.c
index 76169c8..6cdd3ea 100644
--- a/sound/soc/sof/core.c
+++ b/sound/soc/sof/core.c
@@ -8,6 +8,7 @@
  * Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
  */
 
+#include <linux/pm_runtime.h>
 #include <linux/mm.h>
 #include <linux/module.h>
 #include <linux/slab.h>
@@ -331,6 +332,14 @@ static int sof_probe(struct platform_device *pdev)
 			 "warning: failed to initialize trace %d\n", ret);
 	}
 
+	/* autosuspend sof device */
+	pm_runtime_mark_last_busy(sdev->dev);
+	pm_runtime_put_autosuspend(sdev->dev);
+
+	/* autosuspend pci/acpi/spi device */
+	pm_runtime_mark_last_busy(plat_data->dev);
+	pm_runtime_put_autosuspend(plat_data->dev);
+
 	return 0;
 
 comp_err:
-- 
2.7.4

